src/extractor/guidance/turn_analysis.cpp

```diff
@@ -95,6 +97,109 @@ inline bool isRampClass(EdgeID eid, const util::NodeBasedDynamicGraph &node_base
+                    // fork in roundabout
+                    return SPECIAL_EDGEID;
+                }
+                // roundabout does not keep its name
+                if (roundabout_name != 0 && roundabout_name != edge_data.name_id &&
+                    requiresNameAnnounced(name_table.GetNameForID(roundabout_name),
+                                          name_table.GetNameForID(edge_data.name_id)))
+                {
+                    return SPECIAL_EDGEID;
+                }
+
+                roundabout_name = edge_data.name_id;
+
+                continue_edge = edge;
+            }
+            else if (!edge_data.roundabout)
+            {
+                // remember all connected road names
+                connected_names.insert(edge_data.name_id);
+            }
+        }
+        return continue_edge;
+    };
+
+    // the roundabout radius has to be the same for all locations we look at it from
+    // to guarantee this, we search the full roundabout for its vertices
+    // and select the three smalles ids
+    std::vector<NodeID> roundabout_nodes;
+    roundabout_nodes.reserve(40);
+    roundabout_nodes.push_back(nid);
+    while (roundabout_nodes.size() < 40) // hard abort if something is wrong
+    {
+        auto eid = getNextOnRoundabout(roundabout_nodes.back());
+
+        if (eid == SPECIAL_EDGEID)
+        {
+            util::SimpleLogger().Write(logDEBUG) << "Non-Loop Roundabout found.";
+            return false;
+        }
+
+        roundabout_nodes.push_back(node_based_graph.GetTarget(eid));
+
+        if (roundabout_nodes.back() == roundabout_nodes.front()) // circled the roundabout
+        {
+            roundabout_nodes.pop_back();
+            break;
+        }
+    }
+
+    // do we have a dedicated name for the rotary, if not its a roundabout
+    if (roundabout_name == 0 || connected_names.count(roundabout_name))
+    {
+        return false;
+    }
+
+    if (roundabout_nodes.size() <= 1)
+    {
+        return false;
+    }
+
+    std::sort(roundabout_nodes.begin(), roundabout_nodes.end());
+
+    // calculate the radius of the roundabout/rotary. For two coordinates, we assume a minimal
+    // circle
+    // with both vertices right at the other side (so half their distance in meters).
+    // Otherwise, we construct a circle through the first tree vertices.
+    const double radius =
+        roundabout_nodes.size() >= 3
+            ? util::coordinate_calculation::circleRadius(getCoordinate(roundabout_nodes[0]),
+                                                         getCoordinate(roundabout_nodes[1]),
+                                                         getCoordinate(roundabout_nodes[2]))
+            : 0.5 * util::coordinate_calculation::haversineDistance(
+                        getCoordinate(roundabout_nodes[0]), getCoordinate(roundabout_nodes[1]));
+
+    // check whether the circle computation has gone wrong
+    if (std::isinf(radius))
+        return false;
```

Comment:
It can happen if the roundabout-nodes have identical coordinates. We could, instead, do a filtering of the coordinate values based on the coordinates.
Whats your take on this?